astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -24,211 +30,588 @@ class NDArithmeticMixin(object):
+            operands have not-empty meta properties the resulting meta keyword
+            will be operand1 keyword operated upon operand2 keyword. If only
+            one operand has a not-empty meta, the resulting keyword will be
+            the keyword of the operand operated with the ``data`` (this is
+            only allowed if the data is actually a scalar) of the other
+            operand. (There will be more explanation and exampled sometime
+            soon). This parameter will be ignored if ``handle_meta`` is *not*
+            ``True``.
+
+        meta_kwds_set: ``None`` or `dict`
+            If ``None`` there are no keyword settings after the arithmetic
+            meta keyword operations. If this is a `dict` each key/value pair
+            in the ``meta_kwds_set`` will be added (or replaced if the keyword
+            already exists) in the results meta. This parameter will be ignored
+            if ``handle_meta`` is *not* ``True``.
+
+        handle_uncertainties: `bool` or ``None``
+            If ``None`` the result will have no uncertainty. If ``False`` the
+            result will have a copied version of the first operand (or the
+            second if the first had no uncertainty). In case this is ``True``
+            the result will have a correctly propagated uncertainty from the
+            uncertainties of the operands.
+
+        handle_mask: `bool` or ``None``
+            If ``None`` the result will have no mask. If ``False`` the result
+            will have a copied version of the mask of the first operand (or
+            if the first operand has no mask then the one from the second is
+            taken). If ``True`` the masks of both operands will be taken into
+            account and combined (by a bitwise ``or`` operation if the
+            :meth:`NDArithmeticMixin._arithmetic_mask` is not overriden).
+
+        handle_mata: `bool` or ``None``
+            If ``None`` the result will have no meta. If ``False`` the result
+            will have a copied version of the meta of the first operand (or
+            if the first operand has no meta then the one from the second is
+            taken). If ``True`` the meta of both operands will be changed
+            depending on ``meta_kwds_operate`` and ``meta_kwds_set``.
+
+        check_wcs: `bool` or ``None``
+            If ``None`` the result will have no wcs and no comparison between
+            the wcs of the operands is made. If ``False`` the result will have
+            the wcs of the first operand (or second if the first had ``None``).
+            If ``True`` the resulting mask will be like in the case of
+            ``False`` but the wcs information is checked if it is the same
+            for both operands (in some cases it would be bad to apply any
+            arithmetic operation on datasets which have different wcs
+            informations). If they do not match a warning is issues.
```

Comment:
Since astropy.wcs.WCS does not support ``__eq__`` but only ``is`` checks I thought a warning is more appropriate? The documentation also states that ``wcs`` in ``NDData`` is still experimental (http://docs.astropy.org/en/stable/nddata/nddata.html#wcs)